packages/wallet-sdk/src/core/communicator/Communicator.test.ts

```diff
@@ -97,4 +98,122 @@ describe('Communicator', () => {
     );
     expect(mockPopup.postMessage).toHaveBeenNthCalledWith(2, mockRequest, urlOrigin);
   });
+
+  describe('postRPCRequest', () => {
+    const request1 = { id: 'mock-request-id-1-1', data: {} } as unknown as RPCRequestMessage;
+    const request2 = { id: 'mock-request-id-2-2', data: {} } as unknown as RPCRequestMessage;
+    const simulateAsyncCall = () => new Promise((resolve) => setTimeout(resolve, 10));
+
+    let postMessageSpy: jest.SpyInstance;
+    let disconnectSpy: jest.SpyInstance;
+
+    beforeEach(() => {
+      queueMessageEvent(popupLoadedMessage);
+
+      disconnectSpy = jest.spyOn(communicator, 'disconnect');
+      postMessageSpy = jest
```

Comment:
If this test is for Communicator, then Communicator would be the system under test and the methods should avoid being mocked.

High-level options:
- Do nothing: having these tests are better than none
- Mock underlying functions and verify against those calls
- Abstract out `disconnect` and `postMessage` logic and call this `Communicator` something like `MessageHandler` (or keep it the same and abstract out logic)

Reply:
agreed 100%. 
for now, i will just leave it like this as `postMessage` method is fairly simple.
![image](https://github.com/coinbase/coinbase-wallet-sdk/assets/966662/c66819aa-2d7a-47b1-83fb-aa5db4ad5fc5)
will revisit once we have better async behavior on this pr first.